src/Idris/Doc/String.idr

```diff
@@ -244,13 +244,14 @@ getDocsForName fc n
                 | Nothing => pure Empty
            ty <- resugar [] =<< normaliseHoles defs [] (type def)
            let prettyName = pretty (nameRoot nm)
-           let projDecl = hsep [ fun nm prettyName, colon, prettyTerm ty ]
-           let [(_, str)] = lookupName nm (docstrings syn)
-                  | _ => pure projDecl
-           pure $ annotate (Decl nm)
-                $ vcat [ projDecl
-                       , annotate DocStringBody $ vcat (reflowDoc str)
-                       ]
+           let projDecl = annotate (Decl nm) $ hsep [ fun nm prettyName, colon, prettyTerm ty ]
+           case lookupName nm (docstrings syn) of
+                [(_, "")] => pure projDecl
+                [(_, str)] =>
```

Comment:
It seems we always get a docstring back (just that it's sometimes the empty string). I wasn't sure how to best handle missing docstrings. E.g. for interface methods a missing docstring currently result in an empty line in the REPL output. IMHO this way it looks nicer (saves space) but it complicates the code a bit.

Reply:
It looks like we're systematically using `option "" documentation` in the parser.
We should probably refactor this to use `optional documentation` instead and only
bother adding a doc string if there is actually one.

But this will do nicely in the meantime!